sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -603,7 +603,7 @@ size_t coap_opt_put_string(uint8_t *buf, uint16_t lastonum, uint16_t optnum,
 
         part_len = (uint8_t *)uripos - part_start;
 
-        if (part_len) {
```

Comment:
Need to accommodate a request for root path, "/". RFC 7252 sec. 6.4 step 8 says there should not be an option for this path.